Bowling.java

```diff
@@ -2,25 +2,91 @@
 
 public class Bowling {
 
-  private int totalPoints;
+  private Scoreboard scoreboard;
 
   public Bowling() {
-    totalPoints = 0; 
+    scoreboard = new Scoreboard();
+    scoreboard.addFrame();
   }
 
-  public Bowling bowl(int pinsDown) {
-		if (pinsDown < 0) {
-			throw new IllegalArgumentException("Cannot knock down negative pins. Bowl again.");
-		}
-    totalPoints += pinsDown;
+  public Bowling bowl(Integer pinsDown) {
+    if (pinsDown < 0) {
+      throw new IllegalArgumentException("Cannot knock down negative pins. Bowl again.");
+    }
+
+    Frame currentFrame = scoreboard.getCurrentFrame();
```

Comment:
While I agree that the Frame and Scoreboard seem logical, why does Bowling need to know about the Frame at all?  Why does it need to check if the Frame is full?  Since the Scoreboard gave us the Frame can't it just know those details without us having to tell it?  ([Tell Don't Ask](https://pragprog.com/articles/tell-dont-ask) - might be a good read around this idea.)